### PR TITLE
IFNDEF more code that is releated to Filament Autoload

### PR DESCRIPTION
**Pull request by 3d-gussner**

### CODE REVIEW

File: Firmware/ultralcd.cpp [hunk 1]
```diff
@@ -5318,19 +5322,22 @@ static void lcd_main_menu()
...
                 if (fsensor.isEnabled()) {
-                    if (!fsensor.getAutoLoadEnabled()) {
-                        MENU_ITEM_SUBMENU_P(_T(MSG_LOAD_FILAMENT), lcd_LoadFilament);
-                    }
-                    if (fsensor.getFilamentPresent()) {
-                        MENU_ITEM_SUBMENU_P(_T(MSG_UNLOAD_FILAMENT), lcd_unLoadFilament);
-                    }
+//                    if (!fsensor.getAutoLoadEnabled()) {
+//                        MENU_ITEM_SUBMENU_P(_T(MSG_LOAD_FILAMENT), lcd_LoadFilament);
+//                    }
```
Comment (gudnimg) [inline Firmware/ultralcd.cpp]: Dead code?

File: Firmware/Filament_sensor.h [hunk 2]
```diff
@@ -46,8 +46,10 @@ class Filament_sensor {
 
     static void setEnabled(bool enabled);
 
+    #ifndef REMOVE_AUTOLOAD_FILAMENT_MENU_ENTRY
     void setAutoLoadEnabled(bool state, bool updateEEPROM = false);
     bool getAutoLoadEnabled() const { return autoLoadEnabled; }
+    #endif //NOT REMOVE_AUTOLOAD_FILAMENT_MENU_ENTRY
```
Comment (gudnimg) [inline Firmware/Filament_sensor.h:52]: I don't think this is the correct approach, you're disabling too much. These functions are not just used for the Autoload menu item. It's used by `triggerFilamentInserted()` too for normal autoload functionality.

**Review by gudnimg — CHANGES_REQUESTED** [review]
We shouldn't be adding `ifdefs` around `setAutoLoadEnabled` and `getAutoLoadEnabled` because that will affect more than just the Autoload menu item we are looking to disable. The user should still be able to perform autoload by pushing filament into the fsensor.

File: Firmware/Filament_sensor.h (hunk 2 above)
Comment (3d-gussner) [reply Firmware/Filament_sensor.h:52]: Actually these shouldn't be used when `autoload` is disabled. There in the 3rd commit I removed even more code and tested it with MK404
- MK3S
- MK3S + MMU
- MK2.5S

**Comment by 3d-gussner** [comment]
Nevertheless I will close this PR some of the `autoload` features are actually nice to have even when not active.

Gonna keep the branch.

**Pull request closed without merging**